### ISSUE TITLE
Blank screen on iOS when using react-navigation/native-stack

### ISSUE THREAD

**Issue by seyedmostafahasani** [software-mansion/react-native-screens#2405]
### Description

We're using react-navigation/native-stack in our app and react-native-screens version 4.0.0-beta.10. However, when running the iOS app, we encounter a blank screen, while the Android app works correctly with the same configuration.
It works correctly with react-native-screens version 4.0.0-beta.8 on iOS.

### Steps to reproduce

Initialize a new project, then add the `react-navigation` library with all its dependencies. Also, add `react-navigation/native-stack` to handle stacks. If you open the app, you will see a blank screen.

![simulator_screenshot_AD98FE01-8385-4139-9320-0CC938B1A8E7](https://github.com/user-attachments/assets/52c0519e-61b0-47c3-b1ca-9c5a6e314847)

### Snack or a link to a repository

https://github.com/seyedmostafahasani/sample

### Screens version

4.0.0-beta.10

### React Native version

0.75.4

### Platforms

iOS

### JavaScript runtime

None

### Workflow

React Native (without Expo)

### Architecture

Fabric (New Architecture)

### Build type

Debug mode

### Device

iOS simulator

### Acknowledgements

Yes

**Comment by seyedmostafahasani** [comment]
@kkafar 
I've identified the issue in the `RNSScreenStack.mm` file, specifically in the `updateContainer` function:
```
if (!screen.dismissed && screen.controller != nil && screen.activityState != RNSActivityStateInactive) {
      if (pushControllers.count == 0) {
        // first screen on the list needs to be places as "push controller"
        [pushControllers addObject:screen.controller];
      } else {
        if (screen.stackPresentation == RNSScreenStackPresentationPush) {
          [pushControllers addObject:screen.controller];
        } else {
          [modalControllers addObject:screen.controller];
        }
      }
    }
```
When I remove this condition `screen.activityState != RNSActivityStateInactive`, everything works correctly.

**Comment by kkafar** [comment]
@seyedmostafahasani, thanks for detailed report!

I'm aware of the issue since yesterday, most likely https://github.com/software-mansion/react-native-screens/pull/2389 introduced the regression, however I do not know what's the reason yet. Will be looking into it in upcoming hours. 

Meanwhile, please make sure that you're using latest react-navigation v7 and not v6.

**Comment by seyedmostafahasani** [comment]
@kkafar 
Thank you for following up. I will test with react-navigation v7 and let you know the result.

**Comment by kkafar** [comment]
Fixed by #2406

@seyedmostafahasani, please note that screens v4 is meant to be used together with react navigation v7. We won't support react navigation v6 on v4 line of this library.

**Comment by kkafar** [comment]
I've released 4.0.0-beta.12, which contains the fix.

**Comment by seyedmostafahasani** [comment]
@kkafar 
Thanks again! I appreciate you taking the time to fix it. 
I tested with `beta.9` and `react-navigation v7`, and everything works correctly.
